src/lib/build/prepare.rs

```diff
@@ -2,14 +2,15 @@ use build::context::BuildContext;
 use crossbeam::{channel, deque, thread::scope};
 use indexmap::IndexMap;
 use petgraph::{graph::NodeIndex, Direction};
-use retrieve::cache::Binary;
+use retrieve::Binary;
 use std::{thread::sleep, time};
 use util::{errors::Res, graph::Graph};
 
 // A task is an elba subcommand that should only be available from the current root project. Tasks
 // is a list of the Sources of all the tasks needed for this build.
+// TODO: The root is assumed to be local, even for global installation(by `elba install whatever`),
```

Comment:
The main idea here is limiting Cache's job to only handle global caching and give the control of when to interact with global cahce as well as how to build local package and tasks to the stage handlers, which I've not implemented yet.

Reply:
The thing is, by assuming that the root package is local, we're gonna be making `target` folders in the current directory even if we don't need it and we just want to interact with the global cache... I think it would make sense if a stage handler could pass the root `Layout` to this function so we know where to build things and place artifacts if needed.

Reply:
In the code you say that "the root is assumed to be local," even if we run a command like `elba install myfav/package` which has nothing local to speak of. So what would end up happening (as the code is in your branch right now) is that we would assume that the build output of a `elba install`ed package should go in a target folder relative to the current directory. This doesn't really make sense, hence why I wanted to add back in a `Layout` parameter.

Reply:
In the other comment I mentioned the following:

> I think we don't want to codegen the library in this way. There are two cases that users can build a root package without having the source: installing a global executable and installing a global library(for the use in repl). In this case, it's the job of subcommand handler to setup a temporary root folder with the source of target, run build command on it and then install the output into whatever global place(~/.elba/bin or IDRIS_ROOT/libs).

I think this can solve the problem. :D